src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -401,6 +418,102 @@ private CloudResourceManager cloudResourceManager() throws IOException, GeneralS
             .build();
     }
 
+    /**
+     * Create a client to speak to the appengine admin api
+     */
+    private Appengine appengine() throws GeneralSecurityException, IOException {
+        HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+        GoogleCredential credential = GoogleCredential.getApplicationDefault();
+        if (credential.createScopedRequired()) {
+            credential =
+                credential.createScoped(
+                    Collections.singletonList("https://www.googleapis.com/auth/cloud-platform"));
+        }
+
+        return new Appengine.Builder(httpTransport, jsonFactory, credential)
+            .setApplicationName(resourceConfiguration.getApplicationName())
+            .build();
+    }
+
+    /**
+     * Enable Firestore in native mode in an existing project
+     * @param appengine appengine client
+     * @param googleProjectId name of the google project to create the Firestore DB in
+     * @param locationId location of the Firestore DB
+     * @param timeout how long to wait for the creation operation
+     */
+    private static void enableFirestore(final Appengine appengine,
+                                        final String googleProjectId,
+                                        final String locationId,
+                                        final long timeout) throws IOException, InterruptedException {
+        logger.info("Enabling Firestore in project {} in location {}", googleProjectId, locationId);
+        // Create a request object
+        Appengine.Apps.Create createRequest = appengine.apps().create(new Application()
+            .setId(googleProjectId)
+            .setLocationId(locationId)
+            .setDatabaseType(FIRESTORE_DB_TYPE)
+        );
+
+        // Make sure that Firestore is created in the correct project
```

Comment:
haha - so they don't quite build the request right

Reply:
haha, yeah...that made me sad